src/vcpkg/commands.integrate.cpp

```diff
@@ -482,6 +482,67 @@ With a project open, go to Tools->NuGet Package Manager->Package Manager Console
         Checks::exit_success(VCPKG_LINE_INFO);
     }
 
+    static void integrate_zsh(const VcpkgPaths& paths)
+    {
+        const auto home_path = System::get_environment_variable("HOME").value_or_exit(VCPKG_LINE_INFO);
+        const fs::path zshrc_path = fs::path{home_path} / ".zshrc";
+
+        auto& fs = paths.get_filesystem();
+        const fs::path completion_script_path = paths.scripts / "vcpkg_completion.zsh";
+
+        Expected<std::vector<std::string>> maybe_zshrc_content = fs.read_lines(zshrc_path);
+        Checks::check_exit(
+            VCPKG_LINE_INFO, maybe_zshrc_content.has_value(), "Unable to read %s", fs::u8string(zshrc_path));
+
+        std::vector<std::string> zshrc_content = maybe_zshrc_content.value_or_exit(VCPKG_LINE_INFO);
+
+        // How to use bash completions in zsh: https://stackoverflow.com/a/8492043/10162645
+        bool has_autoload_bashcompinit = false;
+        bool has_bashcompinit = false;
+        std::vector<std::string> matches;
+        for (auto&& line : zshrc_content)
+        {
+            std::smatch match;
+            if (std::regex_match(line, match, std::regex{R"###(^source.*scripts/vcpkg_completion.zsh$)###"}))
```

Comment:
The ^$ dance is already implied by regex_match as opposed to regex_search I think. Given the missing $ on 510 it seems like that might not do what you want?

Reply:
Yeah right. I have copied from `integrate_bash`, so I have also fixed it there

Reply:
> I have copied from `integrate_bash`, so I have also fixed it there

Ouch